widget/list.go

```diff
@@ -398,7 +398,52 @@ func (l *List) createWidget() {
 		}...)...)
 		l.container.AddChild(l.hSlider)
 	}
-	l.sliderOpts = nil
+
+}
+
+func (l *List) SetEntries(entries []interface{}) {
+	l.entries = entries
+	l.entriesTable = make(map[interface{}]*Button, len(entries))
+	l.container.RemoveChildren()
+	l.createWidget()
+	l.resetFocusIndex()
+}
+
+func (l *List) RemoveEntry(entry interface{}) {
+	l.init.Do()
+
+	for i, e := range l.entries {
+		if e == entry {
+			l.entries = append(l.entries[:i], l.entries[i+1:]...)
+			but := l.buttons[i]
+			l.container.RemoveChild(but)
+			delete(l.entriesTable, e)
+			l.buttons = append(l.buttons[:i], l.buttons[i+1:]...)
+			break
+		}
+	}
+
+	l.resetFocusIndex()
+}
+
+func (l *List) AddEntry(entry interface{}) {
+	l.init.Do()
+
+	l.entries = append(l.entries, entry)
+	but := NewButton(
```

Comment:
Since we're doing this NewButton in a couple places now, would you please move it to a private func? createEntry(entry) or something similar?